src/history/api.py

```diff
@@ -397,3 +398,45 @@ def pdf_exists(year: str | int,
         bool: True if and only if the PDF file exists.
     """
     return Path.is_file(get_pdf_filepath(year, semester, student_type, round_num))
+
+
+def get_latest_year_and_sem_with_coursereg_info() -> tuple[str, str]:
+    """
+    Returns the latest year/sem that has coursereg PDF data.
+
+    Returns:
+    ----
+        tuple[str, str]: Tuple containing (acad year, sem).
+    """
+
+    curYear = datetime.now().year
+
+    def get_acad_year_starting_this_calendar_year(curYear: int) -> str:
+        """
+        Returns the later AY starting in curYear.
```

Comment:
I think it could be helpful if you added an example here such as `For example, if curYear = 2024, then it returns "2425".`